plutus-core/plutus-ir/src/PlutusIR/Analysis/Builtins.hs

```diff
@@ -1,19 +1,29 @@
 {-# LANGUAGE KindSignatures  #-}
 {-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE RankNTypes #-}
 module PlutusIR.Analysis.Builtins where
 
+import Data.Monoid
 import Control.Lens.TH
 import Data.Kind
 import PlutusCore.Builtin
 import PlutusCore.Builtin qualified as PLC
 import PlutusPrelude (Default (..))
+import PlutusIR qualified as PIR
+
+newtype RewriteRules uni fun = RewriteRules {
```

Comment:
i am not fond of this newtype, but I had to do it to make makeLenses happy.

Also i could make derive-newtype semigroup,monoid but then I had to write custom instances.

Reply:
I'm confused by the makeLenses problem?

Reply:
I would have thought you could just derive semigroup and monoid via `Dual Endo` whatever if that's what you need.

Reply:
Can you try it out? RankNTypes get in the way for me both for makeLenses and for derive-newtype Semigroup,Monoid

Reply:
The lenses seem fine, deriving doesn't work because of the `forall`s. I would either just write the instances by hand or have a list in there and evaluate it where you use it.